src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -569,6 +573,48 @@ protected void validateSequenceDictionaries() {
         }
     }
 
+    /**
+     * Register the reference file (and associated dictionary and index) to be downloaded to every node using Spark's
+     * copying mechanism ({@code SparkContext#addFile()}).
+     * @param ctx the Spark context
+     * @param referenceFile the reference file, can be a local file or a remote path
+     * @return the reference file name; the absolute path of the file can be found by a Spark task using {@code SparkFiles#get()}
+     */
+    protected String addReferenceFilesForSpark(JavaSparkContext ctx, String referenceFile) {
+        Path referencePath = IOUtils.getPath(referenceFile);
+        Path indexPath = ReferenceSequenceFileFactory.getFastaIndexFileName(referencePath);
```

Comment:
I presume this behaves reasonably if the files are empty correct? We aren't restricting our spark files to only running in the case where there exists both a dict and a fai next door.

Reply:
I added a file-existence check for the dict and fai. Also made the method static.